docs/reference/api/usb/USBMSD.md

```diff
@@ -0,0 +1,33 @@
+## USBMSD
+
+<span class="images">![](https://os.mbed.com/docs/v5.9/feature-hal-spec-usb-device-doxy/class_u_s_b_m_s_d.png)<span>USBMSD class hierarchy</span></span>
+
+You can use the USBMSD interface to emulate a mass storage device over USB. You can use this class to store or load data to and from a storage chip, such as SDcard or Flash. This class implements the MSD protocol and takes in a [BlockDevice](blockdevice.html) to interact with a storage chip.
```

Comment:
What are some best practices for using USBMSD? Could you explain the MSD protocol a little more? How does it interact with the storage chip? Why would you want to emulate a mass storage device over USB?